backend/src/src-search/src/main/java/org/eclipse/sw360/search/db/AbstractDatabaseSearchHandler.java

```diff
@@ -58,7 +58,37 @@ public abstract class AbstractDatabaseSearchHandler {
                     "    ret.add(doc.type, {\"field\": \"type\"} );" +
                     "    return ret;" +
                     "}");
-
+    private static final LuceneSearchView luceneFilteredSearchView = new LuceneSearchView("lucene", "restrictedSearch",
+            "function(doc) {" +
+                    "    var ret = new Document();" +
+                    "    if(!doc.type) return ret;" +
+                    "    function idx(obj) {" +
+                    "        for (var key in obj) {" +
+                    "            switch (typeof obj[key]) {" +
+                    "                case 'object':" +
+                    "                    idx(obj[key]);" +
+                    "                    break;" +
+                    "                case 'function':" +
+                    "                    break;" +
+                    "                default:" +
+                    "                    ret.add(obj[key]);" +
+                    "                    break;" +
+                    "            }" +
+                    "        }" +
+                    "    };" +
+                    "    idx(doc);" +
+                    "    ret.add(doc.type, {\"field\": \"type\"} );" +
+                    "    if(doc.name !== undefined && doc.name != null && doc.name.length >0) {  "+
+                    "      ret.add(doc.name, {\"field\": \"name\"} );" +
+                    "    }" +
+                    "    if (doc.fullname !== undefined && doc.fullname != null && doc.fullname.length >0) {  "+
+                    "      ret.add(doc.fullname, {\"field\": \"fullname\"} );" +
+                    "    }" +
+                    "    if (doc.title !== undefined && doc.title != null && doc.title.length >0) {  "+
```

Comment:
`if (doc.title && doc.title.length > 0)` should be enough.

Same applied to `doc.name` at line `81`  and `doc.fulllname` at line `84`.